editors/code/src/inlay_hints.ts

```diff
@@ -71,64 +77,69 @@ class HintsUpdater implements Disposable {
         vscode.window.onDidChangeVisibleTextEditors(
             this.onDidChangeVisibleTextEditors,
             this,
-            this.disposables
+            this.disposables,
         );
 
         vscode.workspace.onDidChangeTextDocument(
             this.onDidChangeTextDocument,
             this,
-            this.disposables
+            this.disposables,
         );
 
         // Set up initial cache shape
-        ctx.visibleRustEditors.forEach(editor => this.sourceFiles.set(
-            editor.document.uri.toString(),
-            {
+        ctx.visibleRustEditors.forEach(editor =>
+            this.sourceFiles.set(editor.document.uri.toString(), {
                 document: editor.document,
                 inlaysRequest: null,
-                cachedDecorations: null
-            }
-        ));
+                cachedDecorations: null,
+            }),
+        );
 
         this.syncCacheAndRenderHints();
     }
 
-    dispose() {
+    dispose(): void {
         this.sourceFiles.forEach(file => file.inlaysRequest?.cancel());
-        this.ctx.visibleRustEditors.forEach(editor => this.renderDecorations(editor, { param: [], type: [] }));
+        this.ctx.visibleRustEditors.forEach(editor =>
+            this.renderDecorations(editor, { param: [], type: [] }),
+        );
         this.disposables.forEach(d => d.dispose());
     }
 
-    onDidChangeTextDocument({ contentChanges, document }: vscode.TextDocumentChangeEvent) {
+    onDidChangeTextDocument({ contentChanges, document }: vscode.TextDocumentChangeEvent): void {
         if (contentChanges.length === 0 || !isRustDocument(document)) return;
         this.syncCacheAndRenderHints();
     }
 
-    private syncCacheAndRenderHints() {
+    private syncCacheAndRenderHints(): void {
         // FIXME: make inlayHints request pass an array of files?
-        this.sourceFiles.forEach((file, uri) => this.fetchHints(file).then(hints => {
-            if (!hints) return;
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        this.sourceFiles.forEach((file, uri) =>
+            this.fetchHints(file).then(hints => {
```

Comment:
```suggestion
            void this.fetchHints(file).then(hints => {
```
Does this silence the misused promises error?

Reply:
Actually, it might. I'll have to check that.

Reply:
Using `void` did work there.